src/qhttpconnection.cpp

```diff
@@ -73,8 +72,9 @@ QHttpConnection::~QHttpConnection()
 
 void QHttpConnection::socketDisconnected()
 {
-    deleteLater();
     invalidateRequest();
+	m_socket->deleteLater();
```

Comment:
Thank you for the patch. Please fix the indentation here.

Also, I do not understand why this segfaults in the dtor? Is it because m_socket is already being freed somewhere else?